### DISCUSSION TITLE
🔥🔥🔥🔥 質問テンプレート 🔥🔥🔥🔥

### DISCUSSION THREAD

**Discussion by txxxxc** [lit-iphone-course/lit-iphone-course#18] [お知らせ]
GitHub Discussionsに質問を投げる時のテンプレートです！
質問を投げる際は以下のフォーマットに従ってやってもらえると講師が喜びます！！
合わせて以下の2つも参照して下さい！
- [質問する時の便利tips
](https://github.com/lit-iphone-course/lit-iphone-course/discussions/18#discussioncomment-3654095)
- [良い質問の例](https://github.com/lit-iphone-course/lit-iphone-course/discussions/18#discussioncomment-4949560)

以下をすべてコピーして書き変える方式がおすすめです！

## 📝タイトルについて📝
質問したいトピックをできるだけ具体的に書いてください！
### 例
```
# 良い例
- Realmに保存した画像のパスを画像に変換できない
- CollectionViewのCell内に、画面を長押ししたら画像が表示されるようにしたい
- 変更したimageviewとlabelのデータをアプリを再起動しても変更した情報が残っているようにしたい

# 悪い例
- Realm ← Realmの何について聞きたいのかが分からない(エラーなのか、仕様なのか)
- ViewControllerのエラー ← ViewControllerのどの部分でエラーが起きているか分からない
```

## (1) 実現したいこと/知りたいこと

(この質問を通して解決したいこと、知りたいことを書いて下さい！)

## (2) 実現するために自分が試した内容とその結果 / 

(エラーの場合、試した挙動とその結果について記入お願いします！)
(概念についての質問の場合は飛ばしてもらって構いません！)

## (3) 調べた内容や資料

(最低2つは調べてURLを貼ってください！)

## (4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
```swift
ここにソースコードを貼る
```
または、ここにGitHubにアップロードされているファイルのリンクを貼る

## (5) GitHubのリンク
エラーが発生しているプロジェクトのURLを貼って下さい！

> **Warning**
> 質問を投稿する前に必ず最新のコードがGitHubにアップロードされていることを確認してください！

**Comment by nokonoko2057** [comment]
# 質問する時の便利tips

## githubのコードの任意の箇所のリンクを取得する
エラーが出た箇所、わからない箇所を選択するのにおすすめ。


1. githubでエラーが出た箇所などの行番号をクリック。行番号をドラッグすると複数行選択できる
1. Copy permalinkをクリック
1. 該当の行へのリンクがコピーされる

![スクリーンショット 2022-09-15 22 42 09](https://user-images.githubusercontent.com/12772107/190419574-8aa04fb7-e7eb-4e33-842e-e800af040a7a.png)

例
https://github.com/lifeistech/toybox/blob/4408f30f1abf28c2ecb88ee24b5bf99e0d722515/UIFont/UIFont/ViewController.swift#L3



## コードにシンタックスハイライトをつける
コードの最初と最後を `3つで囲って、swiftとつけるとシンタックスハイライトがつく

![スクリーンショット 2022-09-15 22 34 45](https://user-images.githubusercontent.com/12772107/190418654-33ff641f-a516-4dcb-89c6-b9ca9bc3b06c.png)

```swift
class ViewController: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        // コメント
    }
}
```

**Comment by txxxxc** [comment]
# 良い質問の例

![image](https://user-images.githubusercontent.com/22112440/218317599-1c7ce345-c8df-4e9d-b901-4ff4ce4ae215.png)

-----

![image](https://user-images.githubusercontent.com/22112440/218317649-d6f94a4f-36f3-4f46-a962-3bde86d538b0.png)

-----

![image](https://user-images.githubusercontent.com/22112440/218317674-c1c14ba5-f4d9-456d-ba9b-8cc06b6b1509.png)

-----

![image](https://user-images.githubusercontent.com/22112440/218317725-1b90c60a-088d-48a5-bcbc-3f543047b701.png)
